### PR TITLE
Add a new optional field to out

### PR DESCRIPTION
**Pull request by rheaton**
- allows for file upload to existing releases
- no other changes to releases are permitted in this commit

Use case: Release is created on Tanzunet, so that external teams
can use the release URL to begin integrations while release files
are being finalized.

Thank you for contributing to the Pivnet Resource!

- Have you made this pull request to the `develop` branch?
- Have you [run the tests locally](https://github.com/pivotal-cf/pivnet-resource#running-the-tests)?

### CODE REVIEW

**Comment by rheaton** [comment]
I don't see a develop branch, please let me know if I should change what branch this is against.

**Review by nedenwalker — COMMENTED** [review]
Notes for Tanzu Network team to review:
1) the method gp/gp.go#UpdateRelease, what does it do if it was determined there was no established way to update a release with files after creation?
2) we cannot update a release if the release has been GAed (made available to all users), so need to ensure that is not violated with this PR

File: out/release/release_finder_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	)
+	BeforeEach(func() {
+		pivnetClient = &releasefakes.FinderClient{}
+		finderErr = nil
+		productSlug = "special-product"
+	})
+
+	Describe("Find", func() {
+
+		Context("", func() {
```
Comment (nedenwalker) [inline out/release/release_finder_test.go:29]: Empty Context?

**Comment by berlin-ab** [comment]
@nedenwalker I was thinking about this feature. Would it be possible to use the version number as the identifier for performing updates? As I understand the rules, the version is unique, and therefore could be used to identify the release. It is also a string that we generate, which would remove the need for us to look up a particular release's identifier by fetching all releases and selecting by the version. cc @rheaton

**Pull request merged**